components/atomic/buttons/ButtonControl/patterns/ButtonControlConfirm.tsx

```diff
@@ -19,28 +19,28 @@ const ButtonControlConfirm = forwardRef(
       children,
       onClick,
       "aria-label": actionLabel,
-      onOpenModal: closeDropdown,
+      closeDropdown,
     }: Props,
     ref
   ) => {
     const dialog = useDialogState({ visible: false, animated: true });
 
+    const handleClick = (e: React.MouseEvent) => {
+      e.preventDefault();
+      if (closeDropdown) {
+        closeDropdown();
+      }
+      dialog.show();
+    };
+
     return (
       <>
-        <ButtonControl
-          as="a"
-          icon={icon}
-          ref={ref}
-          onClick={(e) => {
-            if (closeDropdown) {
-              closeDropdown(e);
-            }
-            dialog.show();
-          }}
-        >
-          {children}
-        </ButtonControl>
-        <Modal label={modalLabel} dialog={dialog}>
+        <DialogDisclosure {...dialog} onClick={handleClick}>
```

Comment:
Updated the `handleClick` and reinserted `DialogDisclosure`. Hopefully this fixes the tab issue!